ctp2_code/gfx/spritesys/director.cpp

```diff
@@ -667,126 +669,126 @@ class DirectorImpl : public Director {
-		MapPoint const &    newPos,
-		sint32              numRevealed,
-		UnitActor **        revealedActors,
-		sint32              arraySize,
-		UnitActor **        moveActors
+		const Unit     &mover,
+		MapPoint const &startPosition,
+		MapPoint const &endPosition,
+		sint32         numberOfRevealedActors,
+		UnitActor      **revealedActors,
+		sint32         numberOfMoveActors,
+		UnitActor      **moveActors
 	);
-	virtual void AddAttack(Unit attacker, Unit attacked);
-	virtual void AddAttackPos(Unit attacker, MapPoint const & pos);
-	virtual void AddSpecialAttack(Unit attacker, Unit attacked, SPECATTACK attack);
-	virtual void AddDeath(UnitActor *dead, const MapPoint &deadPos, sint32 deadSoundID);
+	virtual void AddAttack(const Unit& attacker, const Unit& attacked);
+	virtual void AddAttackPos(const Unit& attacker, const MapPoint &position);
+	virtual void AddSpecialAttack(const Unit& attacker, const Unit& attacked, SPECATTACK attack);
+	virtual void AddDeath(UnitActor *dead, const MapPoint &deadPosition, sint32 deadSoundID);
 	virtual void AddProjectileAttack(
-		Unit shooting,
-		Unit target,
-		SpriteState *projectile_state,
-		SpriteState *projectileEnd_state,
-		sint32 projectile_Path
+		const Unit  &shooting,
+		const Unit  &target,
+		SpriteState *projectileState,
+		SpriteState *projectileEndState,
+		sint32      projectilePath
 	);
-	virtual void AddSpecialEffect(MapPoint &pos, sint32 spriteID, sint32 soundID);
-	virtual void AddMorphUnit(UnitActor *morphingActor, SpriteState *ss, sint32 type,  Unit id);
-	virtual void AddHide(Unit hider);
-	virtual void AddShow(Unit hider);
-	virtual void AddWork(Unit worker);
+	virtual void AddSpecialEffect(const MapPoint &position, sint32 spriteID, sint32 soundID);
+	virtual void AddMorphUnit(UnitActor *morphingActor, SpriteState *spriteState, sint32 type, const Unit &id);
+	virtual void AddHide(const Unit &hider);
+	virtual void AddShow(const Unit &hider);
+	virtual void AddWork(const Unit &worker);
 	virtual void AddFastKill(UnitActor *dead);
-	virtual void AddRemoveVision(const MapPoint &pos, double range);
-	virtual void AddAddVision(const MapPoint &pos, double range);
+	virtual void AddRemoveVision(const MapPoint &position, double range);
+	virtual void AddAddVision(const MapPoint &position, double range);
 	virtual void AddSetVisibility(UnitActor *actor, uint32 visibility);
 	virtual void AddSetOwner(UnitActor *actor, sint32 owner);
 	virtual void AddSetVisionRange(UnitActor *actor, double range);
-	virtual void AddCombatFlash(MapPoint const & pos);
+	virtual void AddCombatFlash(const MapPoint &position);
 	virtual void AddCopyVision();
-	virtual void AddCenterMap(const MapPoint &pos);
+	virtual void AddCenterMap(const MapPoint &position);
 	virtual void AddSelectUnit(uint32 flags);
 	virtual void AddEndTurn();
 	virtual void AddBattle(Battle *battle);
-	virtual void AddPlaySound(sint32 soundID, MapPoint const & pos);
+	virtual void AddPlaySound(sint32 soundID, const MapPoint &position);
```

Comment:
Why is it now 'position' instead of 'pos'? 'pos' is the standard name across the code, whenever you are referring to a position on a map. Basically, this change is like renaming the loop variable 'i' to 'index'.

Reply:
This is a kind of fundamental question related to how I see code quality so the answer is a little longer than usual. I am happy to discuss when you disagree with any of the reasoning.

**Summary**: As I overvalued the `do not use abbreviations`-rule against the `be consistent`-rule. I will revert the change and replace `position` back to `pos`.

**Reasoning**
As developer I like the source code to be as readable as possible. Abbrevations distract from that. Abbreviations need to be interpreted and translated to the real message while reading. This costs energy. One of my statements is that if a book contains as many abbreviations as source code nobody would read it.

With code-completion there is also no need for short variable names. Although I think line length should be limited as well so there is a need for balance.

Abbreviations are not the only element for readability, consistency is another. Consistent code is easier to understand. So as in CTP2 `pos` is used for position everywhere the `Director` code is not consistent any more and violates the `be consistent`-rule which means that I need to revert the change.

BTW: I do not buy your statement that it is the same thing as renaming `i` to `index`. Positions have a functional meaning in CTP2 and therefor have more value than a loop variable. In case `i` would be used for an element that has a functional meaning in the application I would also have renamed it to `index`. I did it a couple of times with `p` -> `player`.

Reply:
It depends on the abbreviation, I even have a "Pos 1" on my keyboard so that should be quite common. In principle the more common it is the easier it is to remember and to understand.

And that is the thing about the variable i, it is so common that it does not need to be introduced. And that's what it has in common with pos. Of course, it is a metaphor, so not every aspects carries over. In fact renaming it to index would indicate that it is not a loop variable. In fact, that it is a one letter variable already indicates that it is used in a loop. So that meaning carries over to others of these kinds. So 'p' is most likely in the code an index into the g_player array. But I think it is quite rare. At least I couldn't find when I was looking for it.

Of course, the variable p may stand for position, player, or point, which is not very specific. And may be confusing. So if 'p' is a pointer on a Player object then it is more appropriately named player. You may also player_ptr.

Reply:
I agree and `pos` is definitely on the edge (here on the good side of the edge). The question is what is the edge. My experience is that it is easier to say no abbreviations than this one is allowed and that one is not.